docs/README.md

```diff
@@ -0,0 +1,142 @@
+
+Integrations write their payload into `stdout` and log into `stderr` (for more information, see [integration specs](https://docs.newrelic.com/docs/integrations/integrations-sdk/file-specifications/integration-executable-file-specifications). The agent reads the payloads, performs some processing, and forwards them to the New Relic platform. 
+
+Integrations can produce different [types of 
+data](https://docs.newrelic.com/docs/integrations/infrastructure-integrations/get-started/understand-use-data-infrastructure-integrations). For a list of available integrations see the [docs 
+site](https://docs.newrelic.com/docs/integrations/host-integrations/host-integrations-list).
+
+To facilitate building your own integrations we provide a [Golang SDK](https://github.com/newrelic/infra-integrations-sdk). 
+
+## Binaries
+
+The New Relic infrastructure agent is composed of different binaries. Besides integrations, there are three executables:
+
+```
+├── newrelic-infra
+├── newrelic-infra-ctl
+└── newrelic-infra-service
+```
+
+### `newrelic-infra-service`
+
+This binary is the daemon managed by the service manager (`systemd`, `upstart`, `init-v`).
+
+Its only purpose is to get safe runtime reload/restart. This is achieved by signaling its child `newrelic-infra`. 
+
+### `newrelic-infra`
+
+This binary owns the whole agent runtime. It can be triggered in stand-alone mode if reload/restart features are not required.  
+
+### `newrelic-infra-ctl`
+
+This is the CLI control command to communicate with the agent daemon.
+
+## Runtime
+
+There's three different runtime steps:
+
+1. Startup
+2. Main runtime
+3. Shutdown
+
+### Startup
+
+#### Connectivity check
+
+The agent performs an initial network connection check against New Relic endpoints before bootstrapping the rest of the runtime.
+
+In case of failure, the agent retries connecting to New Relic till the limit of attempts and time is reached. 
+
+#### Connect
+
+This step attempts to uniquely identify the agent/box.
+
+As hostnames are prone to collision and change, a fingerprint is used to present some host/cloud information that might be valuable for identification, such as hostname, cloud instance id, etc.
+
+The agent retrieves the fingerprinting data and requests a unique identifier to the New Relic identity endpoint. Errors at this point behave as circuit breaker, blocking any data submission to the platform.
+
+In case of failure, the agent retries connecting to New Relic till the limit of attempts and time is reached. This step is run concurrently so it avoids blocking the runtime. 
+
+
+#### Main runtime
+
+The main runtime workflow addresses data processing and submission.
+
+Codebase differentiates different paths for:
+
+- Metrics and events: these share the same workflow, as non dimensional metrics are represented through events.
+- Inventory: stateful data. Might require disk persistence.
```

Comment:
This is a bit terse. Can we elaborate on what stateful means here and why it might require disk persistence?